tests/test_views.py

```diff
@@ -98,3 +122,82 @@ def test_update_view_bad(self):
             n = 0 if i == 1 else 1
             r = self.client.get(f'/update/{self.sites[n].id}/')
             self.assertNotEqual(r.status_code, 200)
+
+    def test_create_view_success(self):
+        '''
+        verify the create view gets a success message and redirects
+        '''
+
+        self.client.login(username=f'testuser0', password=f'testuser0')
+        r = self.client.post('/create/', self.data)
+        self.assertEqual(r.status_code, 302)
+
+        messages = list(get_messages(r.wsgi_request))
+        self.assertEqual(str(messages[0]), 'Successfully created CHIME app.')
+
+    def test_create_view_form_error(self):
+        '''
+        verify the create view gives gives form error with invalid input
+        '''
+
+        self.data['population'] = -1
+        self.client.login(username=f'testuser0', password=f'testuser0')
+        r = self.client.post('/create/', self.data)
+        self.assertEqual(r.status_code, 200)
+        self.assertIn('population', r.context['form'].errors.as_data())
+
+    def test_update_view_success(self):
+        '''
+        verify the update view gets a success message, redirects, and updates db
+        '''
+
+        self.data['population'] = 3500000
+        self.client.login(username=f'testuser0', password=f'testuser0')
+        r = self.client.post(f'/update/{self.sites[0].id}/', self.data)
+        self.assertEqual(r.status_code, 302)
+
+        messages = list(get_messages(r.wsgi_request))
+        self.assertEqual(str(messages[0]), 'Successfully updated CHIME app.')
+
+        site = ChimeSite.objects.get(pk=self.sites[0].id)
+        self.assertEqual(site.population, 3500000)
+
+    def test_update_view_form_error(self):
+        '''
+        verify the update view gives gives form error with invalid input
+        '''
+
+        self.data['population'] = -1
+        self.client.login(username=f'testuser0', password=f'testuser0')
+        r = self.client.post(f'/update/{self.sites[0].id}/', self.data)
+        self.assertEqual(r.status_code, 200)
+        self.assertIn('population', r.context['form'].errors.as_data())
+
+    def test_delete_view_success(self):
+        '''
+        verify the delete view gets a success message and redirects
+        '''
+
+        self.client.login(username=f'testuser0', password=f'testuser0')
+        r = self.client.delete(f'/delete/{self.sites[0].id}/')
+        self.assertEqual(r.status_code, 302)
+
+        messages = list(get_messages(r.wsgi_request))
+        self.assertEqual(str(messages[0]), 'Successfully deleted CHIME app.')
+
+    def test_create_chimesite_view(self):
```

Comment:
I'm not sure what this is testing?  Groups is part of the Django internals.  Is this just seeing if we can instantiate ChimeSiteView()?

Reply:
It’s testing the get_context_data method that coverage was complaining about. Does the view not need to be instantiated to use that method? I can’t look at it at the moment

Reply:
should have probably named the test function differently

Reply:
ok - im going to remove that one test pending a future one that tests the view more directly, and accept this PR.  going forward, you can use make coverage to test coverage (it was in the 80s after this PR iirc).